File: app/src/androidTest/java/org/andstatus/app/util/RawResourceUtils.java
/*
 * Based on the example: 
 * http://stackoverflow.com/questions/4087674/android-read-text-raw-resource-file
 */
package org.andstatus.app.util;

import android.content.Context;
import android.content.res.Resources;
import androidx.annotation.RawRes;
import androidx.test.platform.app.InstrumentationRegistry;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;

public class RawResourceUtils {

    public static String getString(@RawRes int id) throws IOException {
        return new String(getBytes(id, InstrumentationRegistry.getInstrumentation().getContext()),
                StandardCharsets.UTF_8);
    }

    /**
     *  reads resources regardless of their size
     */
    public static byte[] getBytes(@RawRes int id, Context context) throws IOException {
        Resources resources = context.getResources();
        InputStream is = resources.openRawResource(id);

        ByteArrayOutputStream bout = new ByteArrayOutputStream();

        byte[] readBuffer = new byte[4 * 1024];

        try {
            int read;
            do {
                read = is.read(readBuffer, 0, readBuffer.length);
                if(read == -1) {
                    break;
                }
                bout.write(readBuffer, 0, read);
            } while(true);

            return bout.toByteArray();
        } finally {
            is.close();
        }
    }
}

File: app/src/androidTest/kotlin/org/andstatus/app/util/RawResourceUtils.kt
/*
 * Based on the example: 
 * http://stackoverflow.com/questions/4087674/android-read-text-raw-resource-file
 */
package org.andstatus.app.util

import android.content.Context
import androidx.annotation.RawRes
import androidx.test.platform.app.InstrumentationRegistry
import java.io.ByteArrayOutputStream
import java.nio.charset.StandardCharsets

object RawResourceUtils {

    fun getString(@RawRes id: Int): String {
        return String(getBytes(id, InstrumentationRegistry.getInstrumentation().context),
                StandardCharsets.UTF_8)
    }

    /**
     * reads resources regardless of their size
     */
    fun getBytes(@RawRes id: Int, context: Context): ByteArray {
        val resources = context.resources
        val bout = ByteArrayOutputStream()
        val readBuffer = ByteArray(4 * 1024)
        return resources.openRawResource(id).use { inputStream ->
            var read: Int
            do {
                read = inputStream.read(readBuffer, 0, readBuffer.size)
                if (read == -1) {
                    break
                }
                bout.write(readBuffer, 0, read)
            } while (true)
            bout.toByteArray()
        }
    }
}
